test/cql-pytest/test_materialized_view.py

```diff
@@ -227,3 +226,31 @@ def test_is_not_operator_must_be_null(cql, test_keyspace):
                 cql.execute(f"CREATE MATERIALIZED VIEW {test_keyspace}.{mv} AS SELECT * FROM {table} WHERE p IS NOT 42 PRIMARY KEY (p)")
         finally:
             cql.execute(f"DROP MATERIALIZED VIEW IF EXISTS {test_keyspace}.{mv}")
+
+# Test that a view can be altered with synchronous_updates property and that
+# the synchronous updates code path is then reached for such view.
+def test_mv_synchronous_updates(cql, test_keyspace):
+    schema = 'p int, v text, primary key (p)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        def new_view():
+            return new_materialized_view(
+                cql, table, '*', 'v, p', 'v is not null and p is not null'
+            )
+
+        with new_view() as sync_mv, new_view() as async_mv:
```

Comment:
Please also create a third view which has synchronous view enabled right at CREATE TABLE time, to check that code path as well.